src/Stratis.Bitcoin/Base/AsyncWork/AsyncProvider.cs

```diff
@@ -0,0 +1,212 @@
+            }
+
+            loopTask = loopInstance.Run(cancellation, repeatEvery ?? TimeSpan.FromMilliseconds(DefaultLoopRepeatInterval), startAfter).RunningTask;
+
+            // task will continue with onAsyncDelegateUnhandledException if @delegate had unhandled exceptions
+            loopTask.ContinueWith(this.onAsyncLoopUnhandledException, loopInstance, TaskContinuationOptions.OnlyOnFaulted);
+
+            // task will continue with onAsyncDelegateCompleted if @delegate completed or was canceled
+            loopTask.ContinueWith(this.onAsyncLoopCompleted, loopInstance, TaskContinuationOptions.NotOnFaulted);
+
+            return loopInstance;
+        }
+
+        /// <summary>
+        ///  This method is called when delegateTask had an unhandled exception.
+        /// </summary>
+        /// <param name="task">The loop task.</param>
+        /// <param name="state">The <see cref="IAsyncDelegateDequeuer"/> that's run by the delegateTask</param>
+        private void onAsyncLoopUnhandledException(Task task, object state)
+        {
+            AsyncTaskInfo delegateInfo;
+            lock (this.lockAsyncLoops)
+            {
+                if (this.asyncLoops.TryGetValue((IAsyncLoop)state, out delegateInfo))
+                {
+                    IAsyncTaskInfoSetter infoSetter;
+                    infoSetter = (IAsyncTaskInfoSetter)delegateInfo;
+                    infoSetter.Exception = task.Exception.GetBaseException();
+                    infoSetter.Status = task.Status;
+                }
+                else
+                {
+                    // Should never happen.
+                    this.logger.LogError("Cannot find the Async Loop related to the faulted task with Id {0}", task.Id);
+                    return;
+                }
+
+                this.logger.LogError(task.Exception.GetBaseException(), "Unhandled exception for async loop {0} with task Id {1}", delegateInfo.FriendlyName, task.Id);
+            }
+        }
+
+        /// <summary>
+        ///  This method is called when delegateTask completed or was canceled.
+        ///  It removes the loop from the dictionary.
+        /// </summary>
+        /// <param name="task">The loop task.</param>
+        /// <param name="state">The IAsyncDelegate that's run by the delegateTask</param>
+        private void onAsyncLoopCompleted(Task task, object state)
+        {
+            bool removed;
+            lock (this.lockAsyncLoops)
+            {
+                removed = this.asyncLoops.Remove((IAsyncLoop)state);
+            }
+
+            if (removed)
+            {
+                this.logger.LogTrace("Async loop task removed. Id: {0}", task.Id);
+            }
+            else
+            {
+                // Should never happen.
+                this.logger.LogError("Cannot find the async loop task with Id {0}", task.Id);
+            }
+        }
+
+        /// <inheritdoc />
+        public IAsyncQueue<T> CreateAsyncQueue<T>()
```

Comment:
Is htis guy forgotten? or handlonig does not applie here?

Reply:
It cannot have unhandled exception because it has no delegate to trigger.. it's just a queue that the user has to consume by himself